src/openApiCodeGenClients/operation.js

```diff
@@ -0,0 +1,140 @@
+/* Callable operations.
+ */
+import { getContainer } from '@globality/nodule-config/lib';
+import { assign, get } from 'lodash';
+import { buildAdapter, buildHeaders } from '../clients/openapi';
+import buildError, { normalizeError } from '../error';
+import { isRetryableOperation } from '../operation';
+import buildResponse from '../response';
+
+function createParamsWrapper(args) {
+    let paramsWrapper;
+    if (Array.isArray(args)) {
+        // Whilst args is an array we need to convert it to an object
+        paramsWrapper = {
+            params: args.reduce((a, v) => ({ ...a, [v]: v }), {}),
+        };
+    } else {
+        // assuming args is an object (i.e next gen typescript api client generation)
+        paramsWrapper = {
+            params: args,
+        };
+    }
+    return paramsWrapper;
+}
+
+/* Create a new callable operation that return a Promise.
+ */
+export default (
+    axiosInstance,
+    ResourceApi, // The actual resource API class
+    config, // Config including things like baseUrl, timeout, retries etc.
+    context, // Extra context for the operation
+    resourceName, // Name of the resource API
+    operation, // Operation the method on the resource API which is going to be called
+    serviceName, // Name of the service which is being called
+    basePath,
+    isMutation, // flag which describes if the operation is a mutation or not
+    requestMethod, // e.g get, post, put etc.
+    path, // This is the path defined on the spec for the given operation
+) => async (req, args, options) => {
+    // TODO - allow options to be passed in to merge with and override
+    // the requestConfig that gets created here
```

Comment:
Should this still be here? Do we need another ticket for that?

Reply:
removed.